src/lib/utils/concepts.h

```diff
@@ -117,12 +129,38 @@ concept strong_type = is_strong_type_v<T>;
 template <class T>
 concept contiguous_strong_type = strong_type<T> && contiguous_container<T>;
 
+template <typename T, typename Capability>
+concept strong_type_with_capability = requires(T a) {
+                                         { strong_type_has_capability<Capability>(a) } -> std::same_as<std::true_type>;
+                                      };
```

Comment:
Given my suggestion of the static template method in `Strong<>` above, this would become:

```C++
template <typename T, typename Capability>
concept strong_type_with_capability = T::template has_capability<Capability>();
```

Godbolt: https://godbolt.org/z/8sPcc3vbo

Reply:
👍